sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -1426,12 +1526,21 @@ abstract class RoundBase(child: Expression, scale: Expression,
     val ce = child.genCode(ctx)
 
     val evaluationCode = dataType match {
-      case DecimalType.Fixed(_, s) =>
-        s"""
-           |${ev.value} = ${ce.value}.toPrecision(${ce.value}.precision(), $s,
-           |  Decimal.$modeStr(), true);
-           |${ev.isNull} = ${ev.value} == null;
-         """.stripMargin
+      case DecimalType.Fixed(p, s) =>
+        if (_scale >= 0) {
+          s"""
+            ${ev.value} = ${ce.value}.toPrecision(${ce.value}.precision(), $s,
+            Decimal.$modeStr(), true);
+            ${ev.isNull} = ${ev.value} == null;"""
+       } else {
+          s"""
+            ${ev.value} = ${ce.value}.toPrecision(${ce.value}.precision(), $s,
+            Decimal.$modeStr(), true);
+            ${ev.isNull} = ${ev.value} == null;"""
+//          s"""
+//          ${ev.value} = new java.math.BigDecimal(${ce.value}).
+//            setScale(${_scale}, java.math.BigDecimal.${modeStr});"""
```

Comment:
```suggestion
//          ${ev.value} = new Decimal(new java.math.BigDecimal(${ce.value}).
//            setScale(${_scale}, java.math.BigDecimal.${modeStr}));"""
```

Reply:
ah, we should follow the scala code: `new Decimal(${ce.value}. toJavaBigDecimal.setScale...)`

Reply:
@cloud-fan I get the following error: 
org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 56, Column 2: **"toJavaBigDecimal" is neither a method, a field, nor a member class of "org.apache.spark.sql.types.Decimal"**

```
s"""
            ${ev.value} = new Decimal(${ce.value}.toJavaBigDecimal
            .setScale(${_scale}, java.math.BigDecimal.${modeStr}));"""
```

Reply:
This is java code, so parentheses is required when calling a function. try `... toJavaBigDecimal()...`?